staging/src/k8s.io/client-go/applyconfigurations/utils.go

```diff
@@ -1715,3 +1718,7 @@ func ForKind(kind schema.GroupVersionKind) interface{} {
 	}
 	return nil
 }
+
+func NewTypeConverter(scheme *runtime.Scheme) *testing.TypeConverter {
+	return &testing.TypeConverter{Scheme: scheme, TypeResolver: internal.Parser()}
+}
```

Comment:
@jpbetz @liggitt this is leaking testing dependencies into golang binaries

```
grep -r NewTypeConverter staging/ | grep testing
staging/src/k8s.io/sample-apiserver/pkg/generated/applyconfiguration/utils.go:func NewTypeConverter(scheme *runtime.Scheme) *testing.TypeConverter {
staging/src/k8s.io/code-generator/cmd/applyconfiguration-gen/generators/util.go:func NewTypeConverter(scheme *{{.runtimeScheme|raw}}) *{{.testingTypeConverter|raw}} {
staging/src/k8s.io/code-generator/examples/MixedCase/applyconfiguration/utils.go:func NewTypeConverter(scheme *runtime.Scheme) *testing.TypeConverter {
staging/src/k8s.io/code-generator/examples/single/applyconfiguration/utils.go:func NewTypeConverter(scheme *runtime.Scheme) *testing.TypeConverter {
staging/src/k8s.io/code-generator/examples/HyphenGroup/applyconfiguration/utils.go:func NewTypeConverter(scheme *runtime.Scheme) *testing.TypeConverter {
staging/src/k8s.io/code-generator/examples/crd/applyconfiguration/utils.go:func NewTypeConverter(scheme *runtime.Scheme) *testing.TypeConverter {
staging/src/k8s.io/apiextensions-apiserver/examples/client-go/pkg/client/applyconfiguration/utils.go:func NewTypeConverter(scheme *runtime.Scheme) *testing.TypeConverter {
staging/src/k8s.io/apiextensions-apiserver/pkg/client/applyconfiguration/utils.go:func NewTypeConverter(scheme *runtime.Scheme) *testing.TypeConverter {
staging/src/k8s.io/client-go/applyconfigurations/utils.go:func NewTypeConverter(scheme *runtime.Scheme) *testing.TypeConverter {
```

do you have an idea how can we remove this dependency?